### PR TITLE
Points of interest

### PR DESCRIPTION
**Pull request by lucasvanhalst**
migrations: https://gist.github.com/lucasvanhalst/52d1ef816460f8af80d3f4a2e8efa272

change 2 variables at the top `data_path` should be a cloudstorage path and `boom_customer_id` is the customer id of main service of the boom community

### CODE REVIEW

File: src/rogerthat/to/__init__.py [hunk 1]
```diff
@@ -81,10 +82,19 @@ def from_model(cls, m):
...
+    def results(self):
+        raise NotImplementedError
 
     def __init__(self, cursor=None, more=False, results=None):
-        super(PaginatedResultTO, self).__init__(cursor=cursor, more=more, results=results or [])
+        if isinstance(cursor, Cursor):
+            cursor_str = cursor.urlsafe()
+        else:
+            cursor_str = cursor
+        super(PaginatedResultTO, self).__init__(cursor=cursor_str, more=more, results=results or [])
```
Comment (RubenMattan) [inline src/rogerthat/to/__init__.py:97]: i guess that results still needs to be typed property no ?

File: src/oca_imaging/api.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+from os import environ
+
+import requests
+from google.appengine.api.app_identity.app_identity import get_application_id
+from typing import List, Optional
+
+from rogerthat.consts import DEBUG
+from rogerthat.settings import get_server_settings
+
+IMAGING_SERVICE_URL = 'http://localhost:8334' if DEBUG else 'https://oca-dot-%s.ew.r.appspot.com' % get_application_id()
```
Comment (RubenMattan) [inline src/oca_imaging/api.py:26]: should be 'imageresizer-dot'

File: src/mcfw/restapi.py [hunk 3]
```diff
@@ -178,7 +178,7 @@ def write_result(self, result):
                 self.response.set_status(result.status_code)
                 result = serialize_complex_value(result, ErrorResponse, False)
             if DEBUG:
-                self.response.out.write(json.dumps(result, indent=2, sort_keys=True))
+                json.dump(result, self.response.out, indent=2, sort_keys=True, ensure_ascii=False)
```
Comment (RubenMattan) [inline src/mcfw/restapi.py:181]: wouldn't it be safer to keep this the same as production?

File: src/rogerthat/bizz/communities/communities.py [hunk 4]
```diff
@@ -89,12 +94,26 @@ def delete_community(community_id, delete=False):
...
-        community.key.delete()
+        keys = [
+            community.key,
+            CommunityGeoFence.create_key(community_id),
+            CommunityUserStatsHistory.create_key(community_id),
+            CommunityUserStats.create_key(community_id),
+            NewsStream.create_key(community_id),
+        ]
+        keys.extend(NewsGroup.list_by_community_id(community_id).fetch(keys_only=True))
+        # todo properly delete homescreen on firestore as well
```
Comment (RubenMattan) [inline src/rogerthat/bizz/communities/communities.py:114]: todo found

File: src/rogerthat/bizz/maps/poi/__init__.py [hunk 5]
```diff
@@ -0,0 +1,553 @@
...
+    language = get_user_profile(app_user).language
+    app_id = get_app_id_from_app_user(app_user)
+    models = ndb.get_multi([MapConfig.create_key(app_id, POI_TAG),
+                            UserProfileInfo.create_key(app_user)])
+    map_config, user_profile_info = models  # type: MapConfig,  UserProfileInfo
+    response = get_map_response(map_config, user_profile_info, [], add_addresses=False)
+    response.functionalities = [MapFunctionality.CURRENT_LOCATION,
+                                MapFunctionality.SEARCH]
+
+    action_place_types = []
```
Comment (RubenMattan) [inline src/rogerthat/bizz/maps/poi/__init__.py:77]: todo check what action_chips should be active on the POI map

File: src/rogerthat/bizz/maps/poi/poi.py [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+        model.status = POIStatus.VISIBLE if model.has_complete_info else POIStatus.INCOMPLETE
+    else:
+        model.status = POIStatus.INVISIBLE
+    return model
+
+
+def _cleanup_uploaded_files(poi_key):
+    for uploaded_file in UploadedFile.list_by_poi(poi_key):  # type: UploadedFile
+        try:
+            cloudstorage.delete(uploaded_file.cloudstorage_path)
```
Comment (RubenMattan) [inline src/rogerthat/bizz/maps/poi/poi.py:141]: does this cleanup the scaled images as well?

File: src/rogerthat/bizz/maps/poi/poi.py [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+            uploaded_file.key.delete()
```
Comment (RubenMattan) [inline src/rogerthat/bizz/maps/poi/poi.py:142]: do this outside try except

File: src/rogerthat/models/news.py [hunk 8]
```diff
@@ -36,7 +36,11 @@
 
 class MediaType(Enum):
     IMAGE = 'image'
+    IMAGE_360 = 'image_360'
     VIDEO_YOUTUBE = 'video_youtube'
+    VIDEO = 'video'
+    VIDEO_360 = 'video_360'
```
Comment (RubenMattan) [inline src/rogerthat/models/news.py:42]: used?

File: src/rogerthat/models/news.py [hunk 9]
```diff
@@ -36,7 +36,11 @@
 
 class MediaType(Enum):
     IMAGE = 'image'
+    IMAGE_360 = 'image_360'
     VIDEO_YOUTUBE = 'video_youtube'
+    VIDEO = 'video'
```
Comment (RubenMattan) [inline src/rogerthat/models/news.py:41]: used?

File: src/rogerthat/models/news.py [hunk 10]
```diff
@@ -36,7 +36,11 @@
...
+    VIDEO_360 = 'video_360'
+    PDF = 'pdf'
```
Comment (RubenMattan) [inline src/rogerthat/models/news.py:43]: used?

File: src/rogerthat/models/settings.py [hunk 11]
```diff
@@ -54,20 +57,18 @@ def from_value(cls, value):
         return model
 
 
-class ServiceAddress(SyncedName):
+class ServiceLocation(SyncedName):
```
Comment (RubenMattan) [inline src/rogerthat/models/settings.py:60]: can we just rename this property? Does the datastore keep track of internal names in a StructuredProperty?

File: src/rogerthat/settings.py [hunk 12]
```diff
@@ -176,6 +176,10 @@ class ServerSettings(CachedModelMixIn, db.Model):
                                             doc='oca-dot-rogerthat-server.ew.r.appspot.com/api api key',
                                             order=501)
 
+    imagingApiKey = add_meta(db.StringProperty(indexed=False),
```
Comment (RubenMattan) [inline src/rogerthat/settings.py:179]: lets rename to imageResizerApiKey

File: src/solutions/common/bizz/images.py [hunk 13]
```diff
@@ -14,63 +14,156 @@
...
+    elif prefix:
         path = '/%(bucket)s/services/%(service)s/%(prefix)s' % {'bucket': OCA_FILES_BUCKET,
                                                                 'service': service_user.email(),
                                                                 'prefix': prefix}
-        return UploadedFile.list_by_user_and_path(service_user, path)
-    return UploadedFile.list_by_user(service_user)
+        qry = UploadedFile.list_by_user_and_path(service_user, path)
+    else:
+        qry = UploadedFile.list_by_user(service_user)  # type: Iterable[UploadedFile]
+    # TODO: maybe add 3 indexes for this instead / use search index
```
Comment (RubenMattan) [inline src/solutions/common/bizz/images.py:165]: todo found

File: src/solutions/common/bizz/settings.py [hunk 14]
```diff
@@ -274,12 +273,32 @@ def get_service_info(service_user, service_identity):
...
 
 
+def get_media_item_models_from_to(media_items, media_models):
+    # type: (List[MediaItemTO], List[UploadedFile] ) -> List[MediaItem]
+    # media_models should never contain None unless the UploadedFile model was very recently deleted
+    file_models = {f.key.urlsafe(): f for f in media_models}
+    media_list = []
+    for media_item in media_items:
+        if media_item.file_reference:
+            file_model = file_models.get(media_item.file_reference)
```
Comment (RubenMattan) [inline src/solutions/common/bizz/settings.py:283]: does these match? one is urlsafe the other not

File: src/solutions/common/migrations/boom_import.py [hunk 15]
```diff
@@ -0,0 +1,399 @@
...
+    poi_id = ndb.IntegerProperty()  # set when imported as POI
+    data = ndb.JsonProperty()
+    media_mapping = ndb.JsonProperty()  # key: relative file path on disk, value: urlsafe ndb key
+
+    @classmethod
+    def create_key(cls, id):
+        return ndb.Key(cls, id)
+
+
+GEMEENTE_BOOM_POI_ID = 10
```
Comment (RubenMattan) [inline src/solutions/common/migrations/boom_import.py:64]: set for production

File: src/solutions/common/migrations/boom_import.py [hunk 16]
```diff
@@ -0,0 +1,399 @@
...
+    if poi.service_email:
+        parent_key = parent_ndb_key(users.User(poi.service_email), SOLUTION_COMMON)
+    elif poi.poi_id:
+        parent_key = PointOfInterest.create_key(poi.poi_id)
+    file_ids = ndb_allocate_ids(UploadedFile, len(media_list), parent_key)
+    for media, file_id in zip(media_list, file_ids):
+        url = media['relative_url']  # type: str
+        extension = '.' + url.rsplit('.', 1)[1]
+        if poi.service_email:
+            base = '/%(bucket)s/services/%(service)s' % {'bucket': OCA_FILES_BUCKET, 'service': poi.service_email}
```
Comment (RubenMattan) [inline src/solutions/common/migrations/boom_import.py:299]: extra folder for files no?

File: src/solutions/common/models/__init__.py [hunk 17]
```diff
@@ -213,13 +213,7 @@ def solution_message_key(self):
...
 
 class SolutionIdentitySettings(db.Expando):
-    name = db.StringProperty(indexed=False)  # TODO: remove after migration 006
-    phone_number = db.StringProperty(indexed=False)  # TODO: remove after migration 006
-    qualified_identifier = db.StringProperty(indexed=False)  # TODO: remove after migration 006
-    description = db.TextProperty()  # TODO: remove after migration 006
-    address = db.TextProperty()  # TODO: remove after migration 006
-    location = db.GeoPtProperty(indexed=False)  # TODO: remove after migration 006
-
+    name = db.StringProperty(indexed=False)  # TODO: remove usages and use ServiceInfo model instead
```
Comment (RubenMattan) [inline src/solutions/common/models/__init__.py:216]: todo found

File: src/solutions/common/bizz/images.py [hunk 18]
```diff
@@ -14,63 +14,156 @@
...
-        'file_id': file_id,
-        'extension': extension
-    }
+    if reference and reference.kind() == PointOfInterest._get_kind():
+        parent_key = reference
+    else:
+        parent_key = parent_ndb_key(service_user, SOLUTION_COMMON)
+    file_id = ndb_allocate_id(UploadedFile, parent=parent_key)
+    if reference and reference.kind() == PointOfInterest._get_kind():
+        cloudstorage_path = '/%(bucket)s/poi/%(poi_id)d%(prefix)s/%(file_id)d%(extension)s' % {
```
Comment (RubenMattan) [inline src/solutions/common/bizz/images.py:58]: can poi have a prefix?

File: services/imageresizer/app.py [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+logging.basicConfig(format='%(levelname)s %(asctime)s - %(filename)s:%(lineno)s - %(message)s')
+DEBUG = False
+if __name__ == '__main__':
+
+    environ.update({
+        'GAE_APPLICATION': 'e~debug',
+        'GAE_ENV': 'debug',
+        'GAE_SERVICE': 'oca-imageresizer',
+        'GOOGLE_CLOUD_PROJECT': 'debug',
+        'PORT': '9001',
```
Comment (RubenMattan) [inline services/imageresizer/app.py:18]: incorrect port

File: src/solutions/common/migrations/boom_import.py (hunk 15 above)
Comment (lucasvanhalst) [reply src/solutions/common/migrations/boom_import.py:64]: it's the id from the boom db so it's the same

File: src/rogerthat/models/settings.py (hunk 11 above)
Comment (lucasvanhalst) [reply src/rogerthat/models/settings.py:60]: I don't see a reason why they would do that

File: src/solutions/common/models/__init__.py (hunk 17 above)
Comment (lucasvanhalst) [reply src/solutions/common/models/__init__.py:216]: well be my guest its used like 200 times, so not in this pr

File: src/solutions/common/bizz/images.py (hunk 18 above)
Comment (lucasvanhalst) [reply src/solutions/common/bizz/images.py:61]: not yet

File: src/rogerthat/bizz/maps/poi/poi.py (hunk 6 above)
Comment (lucasvanhalst) [reply src/rogerthat/bizz/maps/poi/poi.py]: now it does

File: src/solutions/common/bizz/images.py (hunk 13 above)
Comment (lucasvanhalst) [reply src/solutions/common/bizz/images.py:166]: maybe later

File: src/solutions/common/migrations/boom_import.py (hunk 16 above)
Comment (lucasvanhalst) [reply src/solutions/common/migrations/boom_import.py:299]: idk, does it matter?

Comment (RubenMattan) [reply src/solutions/common/migrations/boom_import.py:299]: i guess the picker in the web won't show any media then no?
It's also cleaner to store them the same way as the dashboard would save them

Comment (RubenMattan) [reply src/solutions/common/migrations/boom_import.py:299]: if no prefix let's dump in a 'media' folder

**Comment by RubenMattan** [comment]
implement search in poi dashboard

**Comment by RubenMattan** [comment]
@lucasvanhalst i guess the usage of ```SearchTag.visible_for_end_user()``` is incorrect. (user will see incomplete POI items) 
I would just nuke this property and use ```SearchTag.poi_status(POIStatus.VISIBLE)```

**Pull request merged**